OpenRA.Mods.Common/Traits/Player/ResourceRenderer.cs

```diff
@@ -0,0 +1,196 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Visualizes the state of the `ResourceLayer`.", " Attach this to the world actor.")]
+	public class ResourceRendererInfo : ITraitInfo, Requires<ResourceLayerInfo>
+	{
+		[FieldLoader.Require]
+		[Desc("Only render these ResourceType Type names.")]
```

Comment:
`Only render these ResourceType names.`? (Or lowercase `type`, but imho that sentence would then still be odd.)